### ISSUE TITLE
PyQt5: Crash when opening .pcd file saved with PyQt4

### ISSUE THREAD

**Issue by rluu** [rluu/pricechartingtool#9]
Opening .pcd files that were created/saved with PyQt4, crashes the interpreter.  It is unknown why this happens, so some investigation is needed.
```
(venv)vagrant@vagrant-ubuntu-trusty-64:/vagrant/src$ ./main.py 
libGL error: failed to load driver: swrast
Traceback (most recent call last):
  File "/vagrant/src/ui.py", line 1525, in _openChart
    unpicklePriceChartDocumentDataFromFile(filename)
  File "/vagrant/src/ui.py", line 4461, in unpicklePriceChartDocumentDataFromFile
    priceChartDocumentData = pickle.load(fh)
ImportError: No module named 'PyQt4'
Aborted (core dumped)
(venv)vagrant@vagrant-ubuntu-trusty-64:/vagrant/src$ 
```

**Comment by rluu** [comment]
The reason why it is giving this error is because, when a Python object is pickled, the type of the object is also saved with the information.  In the pre-port code, I serialized some of the PyQt4 objects.  Thus, in the PyQt5 codebase, it is unable to import/find/load the class that is only in PyQt4.

Writing a converter for .pcd files from PyQt4 to PyQt5 is tricky, as I would need to have both pyqt4 and pyqt5 installed, import them carefully, and also rewrite the `__getstate__` and `__setstate__` for every single class that has a Qt object within it.  This is just about every PCT artifact, and also scaling.  Probably other things too that use QColor.  Basically a lot of writing code.

**Comment by rluu** [comment]
Objects that need porting:
```
from PyQt5.QtGui import QTransform # need to port
from PyQt5.QtGui import QColor # need to port
from PyQt5.QtCore import QPointF # need to port

from PyQt5.QtCore import Qt # (probably not required)
from PyQt5.QtCore import QLineF # (probably not required)
from PyQt5.QtGui import QFont # (not required)
from PyQt5.QtCore import QRectF # (not required)
```

**Comment by rluu** [comment]
After spending a fair amount of time writing conversion classes and methods for the Qt classes that I pickle to file, I realized that there are also default class variables in `PriceBarChartSettings` that use `QColor`.  How would these work if I am initializing values to these `QColor`s from settings and dialogs within the UI?  These class variable values also get pickled to the file.  This problem is getting worse and worse.

At this point, I think it's better to go the route of throwing away all the old `.pcd` files that were created with PyQt4.  I don't have much that I really care about from those files except the TTTA dow time period one.

The issue with going this route and ignoring the problem for now, is that when Qt6 comes out, I will be back in the same situation again.  But the counter-argument is, who knows if Qt6 will ever come out, or if there will be any need for me to port my code to PyQt6.

I have stored my PyQt wrapper classes and how to use them at this gist, just in case I need to revisit this potential solution.  
https://gist.github.com/rluu/d5937c13db4de884bbab
